Client/multiplayer_sa/CMultiplayerSA_DeviceSelection.cpp

```diff
@@ -0,0 +1,157 @@
+        targetName.header.id = paths[i].targetInfo.id;
+        targetName.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_TARGET_NAME;
+        targetName.header.size = sizeof(targetName);
+        const LONG targetNameResult = DisplayConfigGetDeviceInfo(&targetName.header);
+
+        DISPLAYCONFIG_SOURCE_DEVICE_NAME sourceName = {};
+        sourceName.header.adapterId = paths[i].sourceInfo.adapterId;
+        sourceName.header.id = paths[i].sourceInfo.id;
+        sourceName.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_SOURCE_NAME;
+        sourceName.header.size = sizeof(sourceName);
+        const LONG sourceNameResult = DisplayConfigGetDeviceInfo(&sourceName.header);
+        if (targetNameResult == ERROR_SUCCESS && sourceNameResult == ERROR_SUCCESS && targetName.monitorFriendlyDeviceName[0] != '\0')
+        {
+            char gdiDeviceName[std::size(sourceName.viewGdiDeviceName)];
+            char monitorFriendlyDeviceName[std::size(targetName.monitorFriendlyDeviceName)];
+            WideCharToMultiByte(CP_ACP, 0, sourceName.viewGdiDeviceName, -1, gdiDeviceName, static_cast<int>(std::size(gdiDeviceName)), nullptr, nullptr);
+            WideCharToMultiByte(CP_ACP, 0, targetName.monitorFriendlyDeviceName, -1, monitorFriendlyDeviceName,
+                                static_cast<int>(std::size(monitorFriendlyDeviceName)), nullptr, nullptr);
+
+            monitorNames.try_emplace(gdiDeviceName, monitorFriendlyDeviceName);
+        }
+    }
+
+    FreeLibrary(user32Lib);
+    return monitorNames;
+#endif
+}
+
+struct RwSubSystemInfo
+{
+    char name[80];
+};
+
+#define FUNC_rwDeviceSystemRequest 0x7F2AB0
+using rwDeviceSystemRequest = RwSubSystemInfo*(__cdecl*)(RwDevice* device, std::int32_t requestId, RwSubSystemInfo* pOut, void* pInOut, std::int32_t numIn);
+static RwSubSystemInfo* RwEngineGetSubSystemInfo_Hooked(RwSubSystemInfo* subSystemInfo, std::int32_t subSystemIndex)
+{
+    auto rwGlobals = *(RwGlobals**)(0xC97B24);
+    auto rwDeviceSystemRequestFunc = (rwDeviceSystemRequest)(FUNC_rwDeviceSystemRequest);
+    auto result = rwDeviceSystemRequestFunc(&rwGlobals->dOpenDevice, 14, subSystemInfo, nullptr, subSystemIndex);
+    if (!result)
+        return nullptr;
+
+    auto pDxDevice = *(IDirect3D9**)0xC97C20;
+    if (!pDxDevice)
+        return subSystemInfo;
+
+    D3DADAPTER_IDENTIFIER9 identifier;
+    if (FAILED(pDxDevice->GetAdapterIdentifier(subSystemIndex, 0, &identifier)))
+        return subSystemInfo;
+
+    static const auto friendlyNames = GetFriendlyMonitorNamesForDevicePaths();
+
+    // If we can't find the friendly name, either because it doesn't exist or we're on an ancient Windows, fall back to the device name
+    auto it = friendlyNames.find(identifier.DeviceName);
+    if (it != friendlyNames.end())
+    {
+        strncpy_s(subSystemInfo->name, it->second.c_str(), _TRUNCATE);
+    }
+    else
+    {
+        strncpy_s(subSystemInfo->name, identifier.Description, _TRUNCATE);
+    }
```

Comment:
It would be better, but i personally don't know how to easily achieve that. This change improves upon the most often encountered scenario - 1 GPU with multiple screens :)
I do not have the possibility of reproducing the multihead setup scenario, so can't really fix that at this point